spec/graphql/mutations/save_repository_mutation_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe 'saveRepository mutation' do
+  let!(:repository) { create :repository_compound }
+  let(:repository_data) do
+    repo = build :repository
+    data = repo.values.slice(:description, :content_type, :public_access).
+      transform_keys { |k| k.to_s.camelize(:lower) }
+    data['visibility'] = data['publicAccess'] ? 'public' : 'private'
```

Comment:
Instead of generating a new visibility value, I suggest to invert the previous value:
```ruby
// invert the previous public_access value
data['visibility'] = repository.public_access ? 'private' : 'public'
```